### DISCUSSION TITLE
License for the library

### DISCUSSION THREAD

**Discussion by jake-starr-goodrx** [bojand/ghz#458] [General]
The current documentation for the library specify that it's a subject of Apache 2.0 license. https://github.com/bojand/ghz/blob/master/LICENSE. However, two libraries that are used as dependencies are licensed under GPL 3.0. Namely, these are:
github.com/OpenPeeDeeP/depguard v1.1.0 
and 
github.com/denis-tingajkin/go-header v0.4.2. 

Links to the those licenses are below:
https://github.com/OpenPeeDeeP/depguard/blob/v2/LICENSE
https://github.com/denis-tingaikin/go-header/blob/main/LICENSE

To my knowledge, the terms of GPL 3.0 require sources of derivative work also being published under similar terms (GPL 3.0), so any usage of these libraries require both publishing the sources of derivative work and making it free of charge. So any software that uses ghz becomes a subject of GPL 3.0 terms. I wonder if these dependencies are critical for the ghz or can be removed to bring it under Apache 2.0 terms

**Comment by bojand** [comment]
Both of those are dependencies of [golangci-lint](https://github.com/golangci/golangci-lint) and are only in the mod file from the `tools` package which is a traditional (pre 1.24) pattern of [managing tools dependencies](https://go.dev/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module). Neither of those are imported into any actual code so neither should be compiled into the binary. 

Technically this can be removed so the tools are installed manually via script in makefile or some other means.
